wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/brigadier/BrigadierDsl.kt

```diff
@@ -0,0 +1,239 @@
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.boolean(
+    name: String,
+    action: RequiredArgumentAction<S>
+) {
+    val argument = RequiredArgumentBuilder.argument<S, Boolean>(
+        name,
+        BoolArgumentType.bool()
+    )
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds a double with [name] as the parameter name.
+ * @param min the minimum value.
+ * @param max the maximum value.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.double(
+    name: String,
+    min: Double = -Double.MAX_VALUE,
+    max: Double = Double.MAX_VALUE,
+    action: RequiredArgumentAction<S>
+) {
+    val argument = RequiredArgumentBuilder.argument<S, Double>(
+        name,
+        DoubleArgumentType.doubleArg(min, max)
+    )
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds a float with [name] as the parameter name.
+ * @param min the minimum value.
+ * @param max the maximum value.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.float(
+    name: String,
+    min: Float = -Float.MAX_VALUE,
+    max: Float = Float.MAX_VALUE,
+    action: RequiredArgumentAction<S>
+) {
+    val argument = RequiredArgumentBuilder.argument<S, Float>(
+        name,
+        FloatArgumentType.floatArg(min, max)
+    )
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds a integer with [name] as the parameter name.
+ * @param min the minimum value.
+ * @param max the maximum value.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.integer(
+    name: String,
+    min: Int = -Int.MAX_VALUE,
+    max: Int = Int.MAX_VALUE,
+    action: RequiredArgumentAction<S>
+) {
+    val argument = RequiredArgumentBuilder.argument<S, Int>(
+        name,
+        IntegerArgumentType.integer(min, max)
+    )
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds a long with [name] as the parameter name.
+ * @param min the minimum value.
+ * @param max the maximum value.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.long(
+    name: String,
+    min: Long = -Long.MAX_VALUE,
+    max: Long = Long.MAX_VALUE,
+    action: RequiredArgumentAction<S>
+) {
+    val argument = RequiredArgumentBuilder.argument<S, Long>(
+        name,
+        LongArgumentType.longArg(min, max)
+    )
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds an executor to the command tree.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+@JvmName("executesReturn")
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.executes(
+    action: CommandActionReturn<S>
+) {
+    executes(action)
+}
+
+/**
+ * Adds an executor to the command tree.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.executes(
```

Comment:
Having an overload like this doesn't work in some cases like
```kt
executes {
  if (...) {
    functionThatAlsoReturnsAnInt()
  }
}
```
will fail to compile.
I think this should be called something else

Reply:
It seems to fail regardless
```kt
fun a(func: () -> Unit) {
    println("a - Unit")
}

@JvmName("b")
fun a(func: () -> Int) {
    println("a - Int")
}

fun main() {
    a { // Overload resolution ambiguity according to IDEA
        return@a Unit
    }
}
```

Reply:
```kt
a (a@ {
    return@a Unit
} as () -> Unit)
```

IDEA marks this cast as unnecessary but if removed the Kotlin compiler states ambiguity